### PR TITLE
chore: comment out unused component exports (PO pilot)

### PR DESCRIPTION
**Pull request by anniehu4**
### Summary:

See https://czi-edu.slack.com/archives/C0310HJ9WEN/p1652367100603579 -- it'll be easiest for us to merge all of `next` instead of trying to split out only pilot components. Commenting out the Project Overview components in `src/index.ts` (referenced from https://github.com/chanzuckerberg/edu-design-system/pull/1050) until we can verify tree-shaking is set up correctly 

### Test Plan:
everything builds correctly

### CODE REVIEW

File: .storybook/recipes/GlobalHeader/GlobalHeader.tsx [hunk 1]
```diff
@@ -1,24 +1,19 @@
...
-  NotificationList,
-  NotificationListItem,
-} from '../../../src';
+import { Button, Icon, Heading } from '../../../src';
+// Project Overview pilot components -- not yet exported from src/index.ts
+import Avatar from '../../../src/components/Avatar';
+import Header from '../../../src/components/Header';
+import Logo from '../../../src/components/Logo';
+import NavContainer from '../../../src/components/NavContainer';
+import NotificationList from '../../../src/components/NotificationList';
```
Comment (dierat) [inline .storybook/recipes/GlobalHeader/GlobalHeader.tsx:10]: Notifications are in the list for the CAD project (though I don't know if that includes this component specifically).

File: .storybook/recipes/GlobalHeader/GlobalHeader.tsx [hunk 2]
```diff
@@ -115,7 +110,7 @@ export const GlobalHeader = ({ className, ...other }: Props) => {
             isActive={true}
             position={isLarge === false ? 'bottom-left' : undefined}
           >
-            <PopoverHeader
+            <Popover.Header
```
Comment (dierat) [inline .storybook/recipes/GlobalHeader/GlobalHeader.tsx:113]: Nice 👍

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@chanzuckerberg/eds",
-  "version": "0.21.0-alpha.3",
+  "version": "0.21.0-alpha.4",
```
Comment (dierat) [inline package.json:3]: Did you mean to add this?

File: src/index.ts [hunk 4]
```diff
@@ -34,35 +31,53 @@ export { default as LayoutLinelengthContainer } from './components/LayoutLinelen
...
 export { default as Tab } from './components/Tab';
 export { default as Tabs } from './components/Tabs';
 export { default as Tag } from './components/Tag';
 export { default as Text } from './components/Text';
 export { default as TextPassage } from './components/TextPassage';
 export { default as Toast } from './components/Toast';
 export { default as Tooltip } from './components/Tooltip';
-export { default as UtilityNav } from './components/UtilityNav';
-export { default as UtilityNavItem } from './components/UtilityNavItem';
-export * from './hooks';
```
Comment (dierat) [inline src/index.ts:67]: Is `./hooks` only used in Project Overview?

File: .storybook/recipes/GlobalHeader/GlobalHeader.tsx (hunk 1 above)
Comment (anniehu4) [reply .storybook/recipes/GlobalHeader/GlobalHeader.tsx:10]: oh interesting ... I _think_ that's Inline Notifications & Banners and such, whereas this is specifically the Notifications Popover in the left nav footer

File: src/index.ts (hunk 4 above)
Comment (anniehu4) [reply src/index.ts:67]: it's only used in `UtilityNav`, I'll leave a comment

File: .storybook/recipes/GlobalHeader/GlobalHeader.tsx (hunk 1 above)
Comment (dierat) [reply .storybook/recipes/GlobalHeader/GlobalHeader.tsx:10]: Ah, that makes sense, thanks!

**Review by dierat — APPROVED** [review]

**Pull request merged**